flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierUnaligner.java

```diff
@@ -104,12 +104,19 @@ public void processBarrier(CheckpointBarrier barrier, InputChannelInfo channelIn
 			for (final CheckpointableInput input : inputs) {
 				input.checkpointStarted(barrier);
 			}
-			notifyCheckpoint(barrier, 0);
+			notifyCheckpoint(barrier);
 		}
 		if (currentCheckpointId == barrierId) {
 			LOG.debug("{}: Received barrier from channel {} @ {}.", taskName, channelInfo, barrierId);
 
 			if (++numBarriersReceived == numOpenChannels) {
+				if (getNumOpenChannels() > 1) {
+					markAlignmentEnd();
+				}
+				else {
+					// Make sure to report 0 for single channel case
+					markAlignmentEnd(0);
```

Comment:
I would be afraid It would be confusing for the users, "how can there be alignment with single channel?!", if ever this happened to cause metric value to be `1ms` (in case of some machine hiccup).